### ISSUE TITLE
podman pull --arch and --platform values are full ignored

### ISSUE THREAD

**Issue by ssbarnea** [containers/podman#14293]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Apparently you can pass any value to `--arch` when doing a pull as that argument would fully be ignores, making impossible to pull a specific architecture.

**Describe the results you received:**

```shell
$ podman pull --arch xxx quay.io/ansible/python-base:latest
Trying to pull quay.io/ansible/python-base:latest...
Getting image source signatures
Copying blob ...
Copying config sha256:2d9a24aa7d131d57e581b2ffa2f8dfa2e58d12ccfdbc1fc18a0bc1904bf0c05e
Writing manifest to image destination
Storing signatures
2d9a24aa7d131d57e581b2ffa2f8dfa2e58d12ccfdbc1fc18a0bc1904bf0c05e
```

**Describe the results you expected:**

* Failure to pull if that architecture does not exist.
* A mention of the architectures that was pulled, as the user might not always know on which architecture he is running with.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
$ podman version
Client:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18.1
Built:        Thu May  5 21:07:47 2022
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.0.2
API Version:  4.0.2
Go Version:   go1.16.14
Built:        Thu Mar  3 14:58:50 2022
OS/Arch:      linux/arm64
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by ssbarnea** [comment]
Current behavior and even the API is different than docker. Docker uses `docker pull --platform arm64 ...` instead of ``--arch`` and gives proper errors on both cases **unsupported architecture** or **missing architecture**.

```
# incorrect architecture given:
$ docker pull --platform xxx quay.io/ansible/python-base:latest
Error response from daemon: "xxx": unknown operating system or architecture: invalid argument
FAIL: 1 (exit code)

# missing architecture, WARNING displayed but another supported ones was pulled
$ docker pull --platform arm64 quay.io/ansible/python-base:latest
latest: Pulling from ansible/python-base
...
Status: Downloaded newer image for quay.io/ansible/python-base:latest
WARNING: image with reference quay.io/ansible/python-base was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64
quay.io/ansible/python-base:latest
```

**Comment by praveenkumar** [comment]
This looks dup of https://github.com/containers/podman/issues/14271 one.

**Comment by vrothberg** [comment]
Yes, that is a dup of many other dups. GitHub doesn't seem good in showing duplicates.

Please comment here: https://github.com/containers/podman/issues/12682

In my testing docker did not error out when doing `--platform linux/bogus`. `xxx` is interpreted as an OS which is easier to check than the architecture.

In case your version of Docker behaves different, please mention the version.

**Comment by vrothberg** [comment]
Actually, I am going to reopen this issue as it is _slightly_ different.  I will add the warnings as requested :heavy_check_mark:

**Comment by vrothberg** [comment]
Opened https://github.com/containers/common/pull/1052.  With this change, Podman behaves as follows (see the "WARNING: "):
```
podman (main) $ ./bin/podman pull --arch xxx quay.io/ansible/python-base:latest
Trying to pull quay.io/ansible/python-base:latest...
Getting image source signatures
Copying blob 6fa7edf44c66 skipped: already exists  
Copying blob 998ceb85f375 skipped: already exists  
Copying blob f33f743beaa6 skipped: already exists  
Copying blob b0eb4fd5a26a skipped: already exists  
Copying blob 45081c646f63 skipped: already exists  
Copying blob b8ea9b41ba10 skipped: already exists  
Copying blob 902d2d706c71 skipped: already exists  
Copying blob 5a88d10fae3e skipped: already exists  
Copying blob 4ca17c9836ac skipped: already exists  
Copying blob 148a20a0616a skipped: already exists  
Copying blob c978fc766a4b skipped: already exists  
Copying blob fdc12ef68aee skipped: already exists  
Copying config 2d9a24aa7d done  
Writing manifest to image destination
Storing signatures
WARNING: requested architecture "xxx" does not match architecture "amd64" of image 2d9a24aa7d131d57e581b2ffa2f8dfa2e58d12ccfdbc1fc18a0bc1904bf0c05e
2d9a24aa7d131d57e581b2ffa2f8dfa2e58d12ccfdbc1fc18a0bc1904bf0c05e
```

**Comment by vrothberg** [comment]
The issue will be fixed with the next version of Podman (v4.2)
